src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
          List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
          Printf.printf "About to fixup the fd\n%!";
-         ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+         Lwt.catch
```

Comment:
Same comment as above - if none of the unprotected lines can raise an exception, then the code is fine. (I noticed that at least `String.sub` can raise an exception if the substring is out of bounds, but I guess that cannot happen in this case.)